File: PathfinderCombatSimulator.ClassLibrary/DiceManager.cs
namespace PathfinderCombatSimulator
{
    public class DiceManager : IDiceManager
    {
        private Random _rnd = new Random();

        public int AddRolls(IEnumerable<DieRoll> dice)
        {
            int returnable = 0;
            foreach(DieRoll d in dice)
            {
                var rollsLeft = d.DieCount;
                while (rollsLeft > 0)
                {
                    if (d.DieSize.HasValue)
                        returnable += Roll(d.DieSize.Value);

                    rollsLeft--;
                }
            }
            return returnable;  
        }

        public int Roll(int dieSize) =>
            _rnd.Next(1, dieSize);
    }
}

File: CPlusPlus/PathfinderCombatSimulator.ConsoleCpp/DiceManager.cpp
#include "DiceManager.h"
#include "DieRoll.h"
#include <cstdlib>

namespace PathfinderCombatSimulator
{

	int DiceManager::AddRolls(const vector<DieRoll>& dice)
	{
        int returnable = 0;

        for (DieRoll d : dice)
        {
            int rollsLeft = d.DieCount;
            while (rollsLeft > 0)
            {
                if (d.DieSize > 0) {
                    auto justRolled = Roll(d.DieSize);
                    returnable += justRolled;
                }
                rollsLeft--;
            }
        }
        return returnable;
	}

    int DiceManager::Roll(int dieSize)
    {
        return rand() % dieSize + 1;
    }


}

